platform/app/src/net/gpii/app/GpiiActivity.java

```diff
@@ -1,158 +1,591 @@
+
+					for (int i = 0; i < services.size(); i++) {
+						process = services.get(i);
+						String name = process.processName;
+						if (name.equals("net.gpii.app")) {
+							break;
+						}
+					}
+
+					Process.killProcess(process.pid);
+
+				} catch (Throwable e) {
+					e.printStackTrace();
+				}
+			}
+		});
+
+		gpiiUpdateStatusButton.setOnClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				checkGPIIServer();
+			}
+		});
+
+		checkGPIIServer();
+		gpiiStatus.requestFocus();
+
+	}
+
+	protected void checkGPIIServer() {
+		SocketClient sc = new SocketClient();
+		sc.execute(GpiiActivity.this);
+	}
+
+	public void updateStatus(Object status) {
+		if (status == (Object) GpiiActivity.GPII_STATE_RUNNING) {
+			gpiiStatus.setText(R.string.gpii_status_running);
+			gpiiStatus.setBackgroundColor(Color.GREEN);
+			gpiiStatus.setTypeface(null, Typeface.BOLD);
+		} else {
+			gpiiStatus.setText(R.string.gpii_status_not_running);
+			gpiiStatus.setBackgroundColor(Color.RED);
+			gpiiStatus.setTypeface(null, Typeface.BOLD);
+		}
+	}
+
+	class SocketClient extends AsyncTask {
+		private static final String GPII_SERVER_HOST = "0.0.0.0";
+		private static final int GPII_SERVER_PORT = 8081;
+		private static final int CONNECTION_TIMEOUT = 3000;
+
+		@Override
+		protected Object doInBackground(Object... arg0) {
+			GpiiActivity activity = (GpiiActivity) arg0[0];
+			Object result = null;
+
+			try {
+				InetAddress address = InetAddress.getByName(GPII_SERVER_HOST);
+				Socket socket = new Socket();
+
+				socket.connect(
+						new InetSocketAddress(address, GPII_SERVER_PORT),
+						CONNECTION_TIMEOUT);
+
+				if (socket.isConnected()) {
+					result = GpiiActivity.GPII_STATE_RUNNING;
+				}
+
+				socket.close();
+			} catch (Exception ex) {
+				result = GpiiActivity.GPII_STATE_NOT_RUNNING;
+			}
+
+			return result;
+		}
+
+		@Override
+		protected void onPostExecute(Object result) {
+			super.onPostExecute(result);
+			GpiiActivity.this.updateStatus(result);
+		}
+	}
+
+	// Check Android version
+	public void getAndroidVersion() {
+
+		int sdkVersion = Build.VERSION.SDK_INT;
+		String release = Build.VERSION.RELEASE;
+
+		if (sdkVersion < 14 || sdkVersion > 21) {
+
+			Intent browserIntent = new Intent(
+					Intent.ACTION_VIEW,
+					Uri.parse("http://wiki.gpii.net/index.php/GPII_Android_Devices_Compatibility_Table"));
+			startActivity(browserIntent);
+		}
+		// Log.i(TAG, "Android SDK: " + sdkVersion + " (" + release + ")");
+
+		if (sdkVersion > 18) {
+			higherVersionKitKat = true;
+		}
+
+	}
+
+	// Check rooted device
+	private void executeShellCommand(String command) {
+
+		java.lang.Process process;
+
+		try {
+			process = Runtime.getRuntime().exec(command);
+			Log.i(TAG, "Rooted device");
+		} catch (Exception e) {
+			Log.i(TAG, "Not rooted device");
+			Intent browserIntent = new Intent(Intent.ACTION_VIEW,
+					Uri.parse("http://wiki.gpii.net/w/List_of_root_devices"));
```

Comment:
Same comment as before. Also, shouldn't be these static URLs be static variables?